tests/v2/03_api/test_api_dataset_images.py

```diff
@@ -254,24 +265,41 @@ async def test_post_new_image(
     assert "new_attribute" not in res.json()["attributes"].keys()
     assert "new_type" not in res.json()["types"]
 
-    # add ivalid new image
+    # add invalid new images
+    res = await client.post(
+        f"{PREFIX}/project/{project.id}/dataset/{dataset.id}/images/",
+        json=invalid_new_image_1,
+    )
+    assert res.status_code == 422
+    assert "already in" in res.json()["detail"][0]
+
     res = await client.post(
         f"{PREFIX}/project/{project.id}/dataset/{dataset.id}/images/",
-        json=invalid_new_image,
+        json=invalid_new_image_2,
     )
     assert res.status_code == 422
-    debug(res.json())
+    assert "not relative to" in res.json()["detail"][0]
+
     # add new image
     res = await client.post(
         f"{PREFIX}/project/{project.id}/dataset/{dataset.id}/images/",
         json=new_image,
     )
     assert res.status_code == 201
 
+    # add new image (to normalize)
+    res = await client.post(
+        f"{PREFIX}/project/{project.id}/dataset/{dataset.id}/images/",
+        json=new_image_to_normalize,
+    )
+    assert res.status_code == 201
+
     # assert changes
     res = await client.post(
         f"{PREFIX}/project/{project.id}/dataset/{dataset.id}/images/query/"
     )
-    assert res.json()["total_count"] == N + 1
+    assert res.json()["total_count"] == N + 2
     assert "new_attribute" in res.json()["attributes"].keys()
     assert "new_type" in res.json()["types"]
+    # assert zarr_url is normalized
+    assert res.json()["images"][-1]["zarr_url"] == f"{ZARR_DIR}/new_zarr_url_2"
```

Comment:
use normalization function on the left-hand-side, as in
normalization(lhs) == lhs